app/src/main/java/com/io/moviO/searchMovie/SearchMovieFragment.kt

```diff
@@ -0,0 +1,101 @@
+package com.io.moviO.searchMovie
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.widget.doAfterTextChanged
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.Navigation
+import com.io.moviO.R
+import com.io.moviO.databinding.FragmentSearchMovieBinding
+import com.io.moviO.moviesList.MovieListAdapter
+import java.util.*
+import kotlin.concurrent.schedule
+
+private const val QUERY_SEARCH_MIN_CHARS: Int = 2
+
+class SearchMovieFragment : Fragment(R.layout.fragment_search_movie),
+    MovieListAdapter.OnMovieClickedListener {
+    private lateinit var binding: FragmentSearchMovieBinding
+    private var timer = Timer()
+    val viewModel: SearchMovieViewModel by lazy { ViewModelProvider(this).get(SearchMovieViewModel::class.java) }
+    private var adapter = MovieListAdapter(this)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding = FragmentSearchMovieBinding.bind(view)
+
+        binding.searchMovieEt.doAfterTextChanged {
+            timer.cancel()
+            timer = Timer()
+            showLoading(true)
+            timer.schedule(2000) {
+                if (it.toString().length < QUERY_SEARCH_MIN_CHARS) {
+                    viewModel.getLatestMovies()
+                } else {
+                    viewModel.searchMovie(it.toString())
+                }
+            }
+        }
+
+        viewModel.movies.observe(viewLifecycleOwner) {
+            when (it) {
```

Comment:
so you have a showLoading method, but when 

`is SearchMovieViewModel.ViewState.NoResults `-> you manually set `binding.progressBar.visibility = View.INVISIBLE`


on top of the method you can do a general approach to loading visibility:

`showLoading(it is SearchMovieViewModel.ViewState.Loading)`  then for all other cases you don't need to set it to false or manually make it Invisible

Also views now have an extension method isVisible instead of `.visibility = View.INVISIBLE`